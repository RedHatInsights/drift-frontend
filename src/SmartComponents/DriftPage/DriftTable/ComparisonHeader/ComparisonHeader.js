import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Tooltip } from '@patternfly/react-core';
import { ClockIcon, TimesIcon, ExclamationTriangleIcon, ServerIcon, BlueprintIcon } from '@patternfly/react-icons';
import { LongArrowAltUpIcon, LongArrowAltDownIcon, ArrowsAltVIcon } from '@patternfly/react-icons';
import moment from 'moment';

import { ASC, DESC } from '../../../../constants';

import HistoricalProfilesPopover from '../../../HistoricalProfilesPopover/HistoricalProfilesPopover';
import ReferenceSelector from '../ReferenceSelector/ReferenceSelector';

class ComparisonHeader extends Component {
    constructor(props) {
        super(props);
    }

    formatDate = (dateString) => {
        return moment.utc(dateString).format('DD MMM YYYY, HH:mm UTC');
    }

    renderSortButton(sort) {
        let sortIcon;

        if (sort === ASC) {
            sortIcon = <LongArrowAltUpIcon className="active-blue" />;
        }
        else if (sort === DESC) {
            sortIcon = <LongArrowAltDownIcon className="active-blue" />;
        }
        else {
            sortIcon = <ArrowsAltVIcon className="not-active" />;
        }

        return sortIcon;
    }

    async toggleSort(sortType, sort) {
        const { setHistory, toggleFactSort, toggleStateSort } = this.props;

        if (sortType === 'fact') {
            await toggleFactSort(sort);
        } else {
            await toggleStateSort(sort);
        }

        setHistory();
    }

    renderSystemHeaders() {
        const { fetchCompare, masterList, referenceId, removeSystem, selectedHSPIds,
            selectHistoricProfiles, systemIds, updateReferenceId } = this.props;

        let row = [];
        let typeIcon = '';

        masterList.forEach(item => {
            if (item.type === 'system') {
                typeIcon = <Tooltip
                    position='top'
                    content={ <div>System</div> }
                >
                    <ServerIcon/>
                </Tooltip>;
            } else if (item.type === 'baseline') {
                typeIcon = <Tooltip
                    position='top'
                    content={ <div>Baseline</div> }
                >
                    <BlueprintIcon/>
                </Tooltip>;
            } else if (item.type === 'historical-system-profile') {
                typeIcon = <Tooltip
                    position='top'
                    content={ <div>Historical system</div> }
                >
                    <ClockIcon />
                </Tooltip>;
            }

            row.push(
                <th
                    header-id={ item.id }
                    key={ item.id }
                    className={ item.id === referenceId
                        ? 'drift-header reference-header'
                        : `drift-header ${item.type}-header` }
                >
                    <div>
                        <a
                            onClick={ () => removeSystem(item) }
                            className="remove-system-icon"
                            data-ouia-component-type='PF4/Button'
                            data-ouia-component-id={ 'remove-system-button-' + item.id } >
                            <TimesIcon/>
                        </a>
                    </div>
                    <div className='comparison-header'>
                        <div className="drift-header-icon">
                            { typeIcon }
                        </div>
                        <div className="system-name">{ item.display_name }</div>
                        <div className="system-updated-and-reference">
                            <ReferenceSelector
                                updateReferenceId={ updateReferenceId }
                                item={ item }
                                isReference= { item.id === referenceId }
                            />
                            { item.system_profile_exists === false ?
                                <Tooltip
                                    position='top'
                                    content={ <div>System profile does not exist. Please run insights-client on system to upload archive.</div> }
                                >
                                    <ExclamationTriangleIcon color="#f0ab00"/>
                                </Tooltip> : ''
                            }
                            { item.last_updated
                                ? this.formatDate(item.last_updated)
                                : this.formatDate(item.updated)
                            }
                            { item.type === 'system' || item.type === 'historical-system-profile'
                                ? <HistoricalProfilesPopover
                                    system={ item }
                                    systemIds={ systemIds }
                                    referenceId={ referenceId }
                                    fetchCompare={ fetchCompare }
                                    hasCompareButton={ true }
                                    hasMultiSelect={ true }
                                    selectedHSPIds={ selectedHSPIds }
                                    selectHistoricProfiles={ selectHistoricProfiles }
                                />
                                : null
                            }
                        </div>
                    </div>
                </th>
            );
        });

        return row;
    }

    renderHeaderRow() {
        const { factSort, stateSort } = this.props;

        return (
            <tr className="sticky-column-header" data-ouia-component-type='PF4/TableRow' data-ouia-component-id='comparison-table-header-row'>
                <th
                    className="fact-header sticky-column fixed-column-1 pointer"
                    key='fact-header'
                    id={ factSort }
                    onClick={ () => this.toggleSort('fact', factSort) }
                    data-ouia-component-type="PF4/Button"
                    data-ouia-component-id="fact-sort-button"
                >
                    <div className="active-blue">Fact { this.renderSortButton(factSort) }</div>
                </th>
                <th
                    className="state-header sticky-column fixed-column-2 pointer"
                    key='state-header'
                    id={ stateSort || 'disabled' }
                    data-ouia-component-type='PF4/Button'
                    data-ouia-component-id='state-sort-button'
                    onClick={ () => this.toggleSort('state', stateSort) }
                >
                    { stateSort !== ''
                        ? <div className="active-blue">State { this.renderSortButton(stateSort) }</div>
                        : <div>State { this.renderSortButton(stateSort) }</div>
                    }
                </th>
                { this.renderSystemHeaders() }
            </tr>
        );
    }

    render() {
        return (
            <React.Fragment>
                { this.renderHeaderRow() }
            </React.Fragment>
        );
    }
}

ComparisonHeader.propTypes = {
    factSort: PropTypes.string,
    fetchCompare: PropTypes.func,
    masterList: PropTypes.array,
    referenceId: PropTypes.string,
    removeSystem: PropTypes.func,
    stateSort: PropTypes.string,
    systemIds: PropTypes.array,
    toggleFactSort: PropTypes.func,
    toggleStateSort: PropTypes.func,
    updateReferenceId: PropTypes.func,
    setHistory: PropTypes.func,
    selectedHSPIds: PropTypes.array,
    selectHistoricProfiles: PropTypes.func
};

export default ComparisonHeader;
