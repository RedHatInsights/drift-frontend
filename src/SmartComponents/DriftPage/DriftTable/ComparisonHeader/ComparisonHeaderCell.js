import React from 'react';
import PropTypes from 'prop-types';

const ComparisonHeaderCell = ({
    columnWidth,
    classname,
    key,
    id,
    clickFunc,
    ouiaType,
    ouiaId,
    children
}) => {
    return (
        <th
            className={ classname }
            key={ key }
            id={ id }
            onClick={ clickFunc }
            ref={ columnWidth }
            data-ouia-component-type={ ouiaType }
            data-ouia-component-id={ ouiaId }
        >
            {children}
        </th>
    );
};

ComparisonHeaderCell.propTypes = {
    classname: PropTypes.string,
    columnWidth: PropTypes.object,
    key: PropTypes.string,
    id: PropTypes.string,
    clickFunc: PropTypes.func,
    ouiaType: PropTypes.string,
    ouiaId: PropTypes.string,
    children: PropTypes.node
};

export default ComparisonHeaderCell;
