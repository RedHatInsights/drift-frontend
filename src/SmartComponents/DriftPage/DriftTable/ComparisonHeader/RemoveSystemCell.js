import React from 'react';
import PropTypes from 'prop-types';
import { TimesIcon } from '@patternfly/react-icons';

const RemoveSystemCell = ({
    removeSystemFunc,
    item
}) => {
    return (
        <div>
            <a
                onClick={ () => removeSystemFunc(item) }
                className='remove-system-icon'
                data-testid={ `remove-system-button-${item.id}` }
                data-ouia-component-type='PF4/Button'
                data-ouia-component-id={ `remove-system-button-${item.id}` }
            >
                <TimesIcon/>
            </a>
        </div>
    );
};

RemoveSystemCell.propTypes = {
    removeSystemFunc: PropTypes.func,
    item: PropTypes.object
};

export default RemoveSystemCell;
