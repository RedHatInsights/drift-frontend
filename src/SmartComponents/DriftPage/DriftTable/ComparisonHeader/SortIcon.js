import React from 'react';
import PropTypes from 'prop-types';
import { LongArrowAltUpIcon, LongArrowAltDownIcon, ArrowsAltVIcon } from '@patternfly/react-icons';

import { ASC } from '../../../../constants';

const SortIcon = ({
    classname,
    sort,
    type,
    ...props
}) => {
    const Icon = sort && (sort === ASC ? LongArrowAltUpIcon : LongArrowAltDownIcon) || ArrowsAltVIcon;
    const iconClassname = sort ? 'active-blue' : 'not-active';

    return (
        <div className={ classname }>
            { type } <Icon className={ iconClassname } { ...props } />
        </div>
    );
};

SortIcon.propTypes = {
    classname: PropTypes.string,
    sort: PropTypes.string,
    type: PropTypes.string
};

export default SortIcon;
