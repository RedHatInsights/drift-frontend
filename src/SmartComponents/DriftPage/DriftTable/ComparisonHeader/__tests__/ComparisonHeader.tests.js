import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import ComparisonHeader from '../ComparisonHeader';
import fixtures from './ComparisonHeader.fixtures';

describe('ComparisonHeader react-testing-library', () => {
    let props;
    let mockStore;

    beforeEach(() => {
        mockStore = configureStore();
        props = {
            factSort: '',
            fetchCompare: jest.fn(),
            hasHSPReadPermissions: true,
            masterList: [],
            permissions: {
                hspRead: true
            },
            referenceId: undefined,
            isFirstReference: true,
            removeSystem: jest.fn(),
            stateSort: '',
            systemIds: [],
            toggleFactSort: jest.fn(),
            toggleStateSort: jest.fn(),
            updateReferenceId: jest.fn(),
            setHistory: jest.fn(),
            setColumnHeaderWidth: jest.fn()
        };
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render correctly', () => {
        const { asFragment } = render(
            <ComparisonHeader { ...props }/>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render fact sort asc', () => {
        props.factSort = 'asc';
        const { asFragment } = render(
            <ComparisonHeader { ...props }/>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render fact sort desc', () => {
        props.factSort = 'desc';
        const { asFragment } = render(
            <ComparisonHeader { ...props }/>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render state sort asc', () => {
        props.stateSort = 'asc';
        const { asFragment } = render(
            <ComparisonHeader { ...props }/>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render state sort desc', () => {
        props.stateSort = 'desc';
        const { asFragment } = render(
            <ComparisonHeader { ...props }/>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render fact sort desc on click', () => {
        props.factSort = 'asc';
        render(<ComparisonHeader { ...props }/>);

        userEvent.click(screen.getByText('Fact'));
        expect(props.toggleFactSort).toHaveBeenCalledWith('asc');
    });

    it('should render fact sort asc on click', () => {
        props.factSort = 'desc';
        render(<ComparisonHeader { ...props }/>);

        userEvent.click(screen.getByText('Fact'));
        expect(props.toggleFactSort).toHaveBeenCalledWith('desc');
    });

    it('should render state sort none on click', () => {
        props.stateSort = 'desc';
        render(<ComparisonHeader { ...props }/>);

        userEvent.click(screen.getByText('State'));
        expect(props.toggleStateSort).toHaveBeenCalledWith('desc');
    });

    it('should render state sort asc on click', () => {
        props.stateSort = '';
        render(<ComparisonHeader { ...props }/>);

        userEvent.click(screen.getByText('State'));
        expect(props.toggleStateSort).toHaveBeenCalledWith('');
    });

    it('should render state sort desc on click', () => {
        props.stateSort = 'asc';
        render(<ComparisonHeader { ...props }/>);

        userEvent.click(screen.getByText('State'));
        expect(props.toggleStateSort).toHaveBeenCalledWith('asc');
    });

    it.skip('should remove a system', async () => {
        const store = mockStore(props);
        props.masterList = fixtures.masterListSystem;

        render(<MemoryRouter keyLength={ 0 }>
            <Provider store={ store }>
                <ComparisonHeader { ...props }/>
            </Provider>
        </MemoryRouter>);

        await userEvent.click(screen.getByRole('row', {
            name: /remove-system-icon/i
        }));
        expect(store.removeSystem).toHaveBeenCalled();
    });

    it('should render a system, baseline and hsp', () => {
        const store = mockStore(props);
        props.masterList = fixtures.masterListAll;

        const { asFragment } = render(<MemoryRouter keyLength={ 0 }>
            <Provider store={ store }>
                <ComparisonHeader { ...props }/>
            </Provider>
        </MemoryRouter>);

        expect(asFragment()).toMatchSnapshot();
    });

    it('should call setHistory on toggleFactSort', async () => {
        props.factSort = 'desc';
        render(<ComparisonHeader { ...props }/>);

        await userEvent.click(screen.getByText('Fact'));
        expect(props.setHistory).toHaveBeenCalled();
    });

    it('should call setHistory on toggleStateSort', async () => {
        props.stateSort = 'desc';
        render(<ComparisonHeader { ...props }/>);

        await userEvent.click(screen.getByText('State'));
        expect(props.setHistory).toHaveBeenCalled();
    });

    it('should not render HistoricalProfilesPopover with no hspRead permissions', () => {
        props.masterList = fixtures.masterListAll;
        props.permissions.hspRead = false;
        const { asFragment } = render(
            <ComparisonHeader { ...props }/>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});

describe('ComparisonHeader', () => {
    let props;

    beforeEach(() => {
        props = {
            factSort: '',
            fetchCompare: jest.fn(),
            hasHSPReadPermissions: true,
            masterList: [],
            permissions: {
                hspRead: true
            },
            referenceId: undefined,
            isFirstReference: true,
            removeSystem: jest.fn(),
            stateSort: '',
            systemIds: [],
            toggleFactSort: jest.fn(),
            toggleStateSort: jest.fn(),
            updateReferenceId: jest.fn(),
            setHistory: jest.fn()
        };
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should remove a system', () => {
        props.masterList = fixtures.masterListSystem;

        const wrapper = shallow(
            <ComparisonHeader { ...props }/>
        );

        wrapper.find('a').simulate('click');
        expect(props.removeSystem).toHaveBeenCalled();
    });
});
