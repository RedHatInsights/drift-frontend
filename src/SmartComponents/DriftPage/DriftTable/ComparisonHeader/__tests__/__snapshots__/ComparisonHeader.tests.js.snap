// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonHeader should not render HistoricalProfilesPopover with no hspRead permissions 1`] = `
<Fragment>
  <tr
    className="sticky-column-header"
    data-ouia-component-id="comparison-table-header-row"
    data-ouia-component-type="PF4/TableRow"
  >
    <th
      className="fact-header sticky-column fixed-column-1 pointer"
      data-ouia-component-id="fact-sort-button"
      data-ouia-component-type="PF4/Button"
      id=""
      key="fact-header"
      onClick={[Function]}
    >
      <div
        className="active-blue"
      >
        Fact 
        <ArrowsAltVIcon
          className="not-active"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </div>
    </th>
    <th
      className="state-header sticky-column fixed-column-2 pointer right-border"
      data-ouia-component-id="state-sort-button"
      data-ouia-component-type="PF4/Button"
      id="disabled"
      key="state-header"
      onClick={[Function]}
    >
      <div>
        State 
        <ArrowsAltVIcon
          className="not-active"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </div>
    </th>
    <th
      className="drift-header right-border baseline-header"
      header-id="12e4d81b-f1c6-489a-a829-9d96f3a71f53"
      key="12e4d81b-f1c6-489a-a829-9d96f3a71f53"
    >
      <div>
        <a
          className="remove-system-icon"
          data-ouia-component-id="remove-system-button-12e4d81b-f1c6-489a-a829-9d96f3a71f53"
          data-ouia-component-type="PF4/Button"
          onClick={[Function]}
        >
          <TimesIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </a>
      </div>
      <div
        className="comparison-header"
      >
        <div>
          <span
            className="drift-header-icon"
          >
            <Tooltip
              content={
                <div>
                  Baseline
                </div>
              }
              position="top"
            >
              <BlueprintIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>
          </span>
          <span
            className="system-name"
          >
            baselineA
          </span>
        </div>
        <div
          className="system-updated-and-reference"
        >
          <ReferenceSelector
            isReference={false}
            item={
              Object {
                "display_name": "baselineA",
                "id": "12e4d81b-f1c6-489a-a829-9d96f3a71f53",
                "type": "baseline",
                "updated": "2021-04-30T23:53:03.235265Z",
              }
            }
            updateReferenceId={[MockFunction]}
          />
          30 Apr 2021, 23:53 UTC
        </div>
      </div>
    </th>
    <th
      className="drift-header right-border system-header"
      header-id="dc47qffd-09rt-2kw7-8b9b-53f4g716fec5"
      key="dc47qffd-09rt-2kw7-8b9b-53f4g716fec5"
    >
      <div>
        <a
          className="remove-system-icon"
          data-ouia-component-id="remove-system-button-dc47qffd-09rt-2kw7-8b9b-53f4g716fec5"
          data-ouia-component-type="PF4/Button"
          onClick={[Function]}
        >
          <TimesIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </a>
      </div>
      <div
        className="comparison-header"
      >
        <div>
          <span
            className="drift-header-icon"
          >
            <Tooltip
              content={
                <div>
                  System
                </div>
              }
              position="top"
            >
              <ServerIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>
          </span>
          <span
            className="system-name"
          >
            systemA
          </span>
        </div>
        <div
          className="system-updated-and-reference"
        >
          <ReferenceSelector
            isReference={false}
            item={
              Object {
                "display_name": "systemA",
                "id": "dc47qffd-09rt-2kw7-8b9b-53f4g716fec5",
                "type": "system",
                "updated": "2020-11-02T12:41:59.029271Z",
              }
            }
            updateReferenceId={[MockFunction]}
          />
          02 Nov 2020, 12:41 UTC
        </div>
      </div>
    </th>
    <th
      className="drift-header right-border historical-system-profile-header"
      header-id="e3ed3e34-eaa1-45c4-8633-94fef1f678f1"
      key="e3ed3e34-eaa1-45c4-8633-94fef1f678f1"
    >
      <div>
        <a
          className="remove-system-icon"
          data-ouia-component-id="remove-system-button-e3ed3e34-eaa1-45c4-8633-94fef1f678f1"
          data-ouia-component-type="PF4/Button"
          onClick={[Function]}
        >
          <TimesIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </a>
      </div>
      <div
        className="comparison-header"
      >
        <div>
          <span
            className="drift-header-icon"
          >
            <Tooltip
              content={
                <div>
                  Historical system
                </div>
              }
              position="top"
            >
              <ClockIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>
          </span>
          <span
            className="system-name"
          >
            hspA
          </span>
        </div>
        <div
          className="system-updated-and-reference"
        >
          <ReferenceSelector
            isReference={false}
            item={
              Object {
                "display_name": "hspA",
                "id": "e3ed3e34-eaa1-45c4-8633-94fef1f678f1",
                "system_id": "dc47qffd-09rt-2kw7-8b9b-53f4g716fec5",
                "type": "historical-system-profile",
                "updated": "2021-05-15T07:24:01+00:00",
              }
            }
            updateReferenceId={[MockFunction]}
          />
          15 May 2021, 07:24 UTC
        </div>
      </div>
    </th>
  </tr>
</Fragment>
`;

exports[`ComparisonHeader should render a system, baseline and hsp 1`] = `
<Fragment>
  <tr
    className="sticky-column-header"
    data-ouia-component-id="comparison-table-header-row"
    data-ouia-component-type="PF4/TableRow"
  >
    <th
      className="fact-header sticky-column fixed-column-1 pointer"
      data-ouia-component-id="fact-sort-button"
      data-ouia-component-type="PF4/Button"
      id=""
      key="fact-header"
      onClick={[Function]}
    >
      <div
        className="active-blue"
      >
        Fact 
        <ArrowsAltVIcon
          className="not-active"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </div>
    </th>
    <th
      className="state-header sticky-column fixed-column-2 pointer right-border"
      data-ouia-component-id="state-sort-button"
      data-ouia-component-type="PF4/Button"
      id="disabled"
      key="state-header"
      onClick={[Function]}
    >
      <div>
        State 
        <ArrowsAltVIcon
          className="not-active"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </div>
    </th>
    <th
      className="drift-header right-border baseline-header"
      header-id="12e4d81b-f1c6-489a-a829-9d96f3a71f53"
      key="12e4d81b-f1c6-489a-a829-9d96f3a71f53"
    >
      <div>
        <a
          className="remove-system-icon"
          data-ouia-component-id="remove-system-button-12e4d81b-f1c6-489a-a829-9d96f3a71f53"
          data-ouia-component-type="PF4/Button"
          onClick={[Function]}
        >
          <TimesIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </a>
      </div>
      <div
        className="comparison-header"
      >
        <div>
          <span
            className="drift-header-icon"
          >
            <Tooltip
              content={
                <div>
                  Baseline
                </div>
              }
              position="top"
            >
              <BlueprintIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>
          </span>
          <span
            className="system-name"
          >
            baselineA
          </span>
        </div>
        <div
          className="system-updated-and-reference"
        >
          <ReferenceSelector
            isReference={false}
            item={
              Object {
                "display_name": "baselineA",
                "id": "12e4d81b-f1c6-489a-a829-9d96f3a71f53",
                "type": "baseline",
                "updated": "2021-04-30T23:53:03.235265Z",
              }
            }
            updateReferenceId={[MockFunction]}
          />
          30 Apr 2021, 23:53 UTC
        </div>
      </div>
    </th>
    <th
      className="drift-header right-border system-header"
      header-id="dc47qffd-09rt-2kw7-8b9b-53f4g716fec5"
      key="dc47qffd-09rt-2kw7-8b9b-53f4g716fec5"
    >
      <div>
        <a
          className="remove-system-icon"
          data-ouia-component-id="remove-system-button-dc47qffd-09rt-2kw7-8b9b-53f4g716fec5"
          data-ouia-component-type="PF4/Button"
          onClick={[Function]}
        >
          <TimesIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </a>
      </div>
      <div
        className="comparison-header"
      >
        <div>
          <span
            className="drift-header-icon"
          >
            <Tooltip
              content={
                <div>
                  System
                </div>
              }
              position="top"
            >
              <ServerIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>
          </span>
          <span
            className="system-name"
          >
            systemA
          </span>
        </div>
        <div
          className="system-updated-and-reference"
        >
          <ReferenceSelector
            isReference={false}
            item={
              Object {
                "display_name": "systemA",
                "id": "dc47qffd-09rt-2kw7-8b9b-53f4g716fec5",
                "type": "system",
                "updated": "2020-11-02T12:41:59.029271Z",
              }
            }
            updateReferenceId={[MockFunction]}
          />
          02 Nov 2020, 12:41 UTC
          <withRouter(Connect(HistoricalProfilesPopover))
            fetchCompare={[MockFunction]}
            hasCompareButton={true}
            hasMultiSelect={true}
            system={
              Object {
                "display_name": "systemA",
                "id": "dc47qffd-09rt-2kw7-8b9b-53f4g716fec5",
                "type": "system",
                "updated": "2020-11-02T12:41:59.029271Z",
              }
            }
            systemIds={Array []}
            systemName="systemA"
          />
        </div>
      </div>
    </th>
    <th
      className="drift-header right-border historical-system-profile-header"
      header-id="e3ed3e34-eaa1-45c4-8633-94fef1f678f1"
      key="e3ed3e34-eaa1-45c4-8633-94fef1f678f1"
    >
      <div>
        <a
          className="remove-system-icon"
          data-ouia-component-id="remove-system-button-e3ed3e34-eaa1-45c4-8633-94fef1f678f1"
          data-ouia-component-type="PF4/Button"
          onClick={[Function]}
        >
          <TimesIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        </a>
      </div>
      <div
        className="comparison-header"
      >
        <div>
          <span
            className="drift-header-icon"
          >
            <Tooltip
              content={
                <div>
                  Historical system
                </div>
              }
              position="top"
            >
              <ClockIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>
          </span>
          <span
            className="system-name"
          >
            hspA
          </span>
        </div>
        <div
          className="system-updated-and-reference"
        >
          <ReferenceSelector
            isReference={false}
            item={
              Object {
                "display_name": "hspA",
                "id": "e3ed3e34-eaa1-45c4-8633-94fef1f678f1",
                "system_id": "dc47qffd-09rt-2kw7-8b9b-53f4g716fec5",
                "type": "historical-system-profile",
                "updated": "2021-05-15T07:24:01+00:00",
              }
            }
            updateReferenceId={[MockFunction]}
          />
          15 May 2021, 07:24 UTC
          <withRouter(Connect(HistoricalProfilesPopover))
            fetchCompare={[MockFunction]}
            hasCompareButton={true}
            hasMultiSelect={true}
            system={
              Object {
                "display_name": "hspA",
                "id": "e3ed3e34-eaa1-45c4-8633-94fef1f678f1",
                "system_id": "dc47qffd-09rt-2kw7-8b9b-53f4g716fec5",
                "type": "historical-system-profile",
                "updated": "2021-05-15T07:24:01+00:00",
              }
            }
            systemIds={Array []}
            systemName="hspA"
          />
        </div>
      </div>
    </th>
  </tr>
</Fragment>
`;

exports[`ComparisonHeader should render correctly 1`] = `
<Fragment>
  <tr
    className="sticky-column-header"
    data-ouia-component-id="comparison-table-header-row"
    data-ouia-component-type="PF4/TableRow"
  >
    <th
      className="fact-header sticky-column fixed-column-1 pointer"
      data-ouia-component-id="fact-sort-button"
      data-ouia-component-type="PF4/Button"
      id=""
      key="fact-header"
      onClick={[Function]}
    >
      <div
        className="active-blue"
      >
        Fact 
        <ArrowsAltVIcon
          className="not-active"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </div>
    </th>
    <th
      className="state-header sticky-column fixed-column-2 pointer right-border"
      data-ouia-component-id="state-sort-button"
      data-ouia-component-type="PF4/Button"
      id="disabled"
      key="state-header"
      onClick={[Function]}
    >
      <div>
        State 
        <ArrowsAltVIcon
          className="not-active"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </div>
    </th>
  </tr>
</Fragment>
`;

exports[`ComparisonHeader should render fact sort asc 1`] = `
<Fragment>
  <tr
    className="sticky-column-header"
    data-ouia-component-id="comparison-table-header-row"
    data-ouia-component-type="PF4/TableRow"
  >
    <th
      className="fact-header sticky-column fixed-column-1 pointer"
      data-ouia-component-id="fact-sort-button"
      data-ouia-component-type="PF4/Button"
      id="asc"
      key="fact-header"
      onClick={[Function]}
    >
      <div
        className="active-blue"
      >
        Fact 
        <LongArrowAltUpIcon
          className="active-blue"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </div>
    </th>
    <th
      className="state-header sticky-column fixed-column-2 pointer right-border"
      data-ouia-component-id="state-sort-button"
      data-ouia-component-type="PF4/Button"
      id="disabled"
      key="state-header"
      onClick={[Function]}
    >
      <div>
        State 
        <ArrowsAltVIcon
          className="not-active"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </div>
    </th>
  </tr>
</Fragment>
`;

exports[`ComparisonHeader should render fact sort desc 1`] = `
<Fragment>
  <tr
    className="sticky-column-header"
    data-ouia-component-id="comparison-table-header-row"
    data-ouia-component-type="PF4/TableRow"
  >
    <th
      className="fact-header sticky-column fixed-column-1 pointer"
      data-ouia-component-id="fact-sort-button"
      data-ouia-component-type="PF4/Button"
      id="desc"
      key="fact-header"
      onClick={[Function]}
    >
      <div
        className="active-blue"
      >
        Fact 
        <LongArrowAltDownIcon
          className="active-blue"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </div>
    </th>
    <th
      className="state-header sticky-column fixed-column-2 pointer right-border"
      data-ouia-component-id="state-sort-button"
      data-ouia-component-type="PF4/Button"
      id="disabled"
      key="state-header"
      onClick={[Function]}
    >
      <div>
        State 
        <ArrowsAltVIcon
          className="not-active"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </div>
    </th>
  </tr>
</Fragment>
`;

exports[`ComparisonHeader should render state sort asc 1`] = `
<Fragment>
  <tr
    className="sticky-column-header"
    data-ouia-component-id="comparison-table-header-row"
    data-ouia-component-type="PF4/TableRow"
  >
    <th
      className="fact-header sticky-column fixed-column-1 pointer"
      data-ouia-component-id="fact-sort-button"
      data-ouia-component-type="PF4/Button"
      id=""
      key="fact-header"
      onClick={[Function]}
    >
      <div
        className="active-blue"
      >
        Fact 
        <ArrowsAltVIcon
          className="not-active"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </div>
    </th>
    <th
      className="state-header sticky-column fixed-column-2 pointer right-border"
      data-ouia-component-id="state-sort-button"
      data-ouia-component-type="PF4/Button"
      id="asc"
      key="state-header"
      onClick={[Function]}
    >
      <div
        className="active-blue"
      >
        State 
        <LongArrowAltUpIcon
          className="active-blue"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </div>
    </th>
  </tr>
</Fragment>
`;

exports[`ComparisonHeader should render state sort desc 1`] = `
<Fragment>
  <tr
    className="sticky-column-header"
    data-ouia-component-id="comparison-table-header-row"
    data-ouia-component-type="PF4/TableRow"
  >
    <th
      className="fact-header sticky-column fixed-column-1 pointer"
      data-ouia-component-id="fact-sort-button"
      data-ouia-component-type="PF4/Button"
      id=""
      key="fact-header"
      onClick={[Function]}
    >
      <div
        className="active-blue"
      >
        Fact 
        <ArrowsAltVIcon
          className="not-active"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </div>
    </th>
    <th
      className="state-header sticky-column fixed-column-2 pointer right-border"
      data-ouia-component-id="state-sort-button"
      data-ouia-component-type="PF4/Button"
      id="desc"
      key="state-header"
      onClick={[Function]}
    >
      <div
        className="active-blue"
      >
        State 
        <LongArrowAltDownIcon
          className="active-blue"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      </div>
    </th>
  </tr>
</Fragment>
`;
